### DISCUSSION TITLE
create admin user

### DISCUSSION THREAD

**Discussion by davidsean** [specify/specify7#3116] [Announcements]
I am trying out the evaluation docker stack and can spin up the service without any issues (after manualy creating/applying the migratations.

However, I am not sure how to create the initial user, as this build does is not seeded with a Specify6 database.

Things I tried:
 - I've tried a shell into the `specify7-specify7` and ran `python manage.py createsuperuser` without success (it's not implemented). 
 - I have manually added a user row in the database and added the password hash. This can get me to log-in, but I then have a problem regarding no existing collections.

**Comment by maxpatiiuk** [comment]
Hi,

Usage of Specify 7 requires a database to be created using Specify 6's Setup Wizard.
There is no [Specify 7 Setup Wizard](https://github.com/specify/specify7/issues/2932) yet.

Thus, you need to [install Specify 6](https://www.specifysoftware.org/join/download/), open SpWizard, and create a database. In SpWizard, you can directly connect to your production MySQL server, or use a local MySQL server, followed by exporting the database dump and importing it on the production machine (this allows to avoid installing specify 6 & java on the production machine)

If you need more documentation of these steps, I will be happy to provide some.
Also, if you are just doing evaluation, I can provide a demo database dump that you can import and use with Specify 7

**Reply by davidsean** [reply]
I will try installing Specify6 and point it to the Specify7's mariaDB container (and pass along credentials). I'll let you know if I need a dump.

**Reply by maxpatiiuk** [reply]
Note, depending on which docker composition of Specify 7 you are using, MariaDB container might not be accessible outside the container.

To make it accessible, add the following to the mariadb section in your docker-compose (unless it's already there):

```yaml
    ports:
      - "127.0.0.1:3306:3306"
```

**Reply by davidsean** [reply]
In case someone tries, I had issues with missing tables after trying to log in Specify7 that used a Specify6 DB (using the wizard).
the fix is to open a shell to the django webserver and run the `permissions` migrations:
``` bash
$ ve/bin/python manage.py makemigrations permissions
$ ve/bin/python manage.py migrate permissions
```
I can now log in the Specify7 web app, ther is still an error message about incompatible schema, but there are instructions on how to fix it (that I've yet tried). At least there is now an interface to play around with.

**Reply by maxpatiiuk** [reply]
The commands above would only run some migrations. To avoid issues, all migrations need to be run. You can do that by executing `make django_migrations` (or directly `ve/bin/python manage.py migrate`)
